widgets/serializers/rss.py

```diff
@@ -31,33 +31,35 @@ class RssFeedWidgetSerializer(WidgetInstanceSerializer):
     name = "RSS Feed"
     description = "RSS Feed"
 
-    def get_html(self, instance):
+    def get_json(self, instance):
         """Renders the widget to html based on configuration"""
-        feed = feedparser.parse(instance.configuration["url"]).entries
-        if not feed:
-            return (
-                "<p>No RSS entries found. You may have selected an invalid RSS url.</p>"
-            )
+        rss = feedparser.parse(instance.configuration["url"])
+        if not rss:
+            return {"title": "RSS", "entries": []}
+
         timestamp_key = (
-            "published_parsed" if "published_parsed" in feed[0] else "updated_parsed"
+            "published_parsed"
+            if rss.entries and "published_parsed" in rss.entries[0]
+            else "updated_parsed"
+        )
+        sorted_feed = sorted(
+            rss.entries, reverse=True, key=lambda entry: entry[timestamp_key]
         )
-        sorted_feed = sorted(feed, reverse=True, key=lambda entry: entry[timestamp_key])
         display_limit = min(
             instance.configuration["feed_display_limit"], MAX_FEED_ITEMS
         )
-        formatted_items = []
-        for entry in sorted_feed[:display_limit]:
-            entry_title = entry.get("title", None)
-            entry_link = entry.get("link", None)
-            entry_timestamp = entry.get(timestamp_key, None)
-            if entry_timestamp:
-                entry_timestamp = time.strftime(TIMESTAMP_FORMAT, entry_timestamp)
-            formatted_items.append(
-                format_html(
-                    '<p><a href="{entry_link}">{entry_timestamp} | {entry_title}<a><p>',
-                    entry_link=entry_link,
-                    entry_timestamp=entry_timestamp,
-                    entry_title=entry_title,
-                )
-            )
-        return "\n".join(formatted_items)
+
+        return {
+            "title": rss.feed.title,
```

Comment:
I think the RSS widget title entered by the user should be used if not blank?

Reply:
:+1: